scantool/diag_os_win.c

```diff
@@ -240,6 +240,51 @@ int diag_os_ipending(void) {
 	return 0;
 }
 
+//diag_os_cursor_up : move the console cursor up the specified number of 
+//lines and to column 1
+void diag_os_cursor_up(int lines) {
+	HANDLE console;
+	CONSOLE_SCREEN_BUFFER_INFO buffer_info;
+	int rv;
+
+	console = GetStdHandle(STD_OUTPUT_HANDLE);
+	if ((console == INVALID_HANDLE_VALUE) || (console == NULL)) {
+		return;
+	}
+	rv = GetConsoleScreenBufferInfo(console, &buffer_info);
+	if (rv == 0) {
+		return;
+	}
+	buffer_info.dwCursorPosition.Y -= lines;
+	if (buffer_info.dwCursorPosition.Y < 0) {
```

Comment:
this will probably not work, `dw` prefix usually means DWORD a.k.a  uint32

Reply:
dwCursorPosition is a COORD, dwCursorPosition.Y is a SHORT, SHORT has range -32768 through 32767

https://learn.microsoft.com/en-us/windows/console/console-screen-buffer-info-str
https://learn.microsoft.com/en-us/windows/console/coord-str
https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types

Reply:
Ah, well thanks microsoft for using useless and misleading prefixes.